components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/STSApplicationMgtListener.java

```diff
@@ -69,6 +71,60 @@ public int getDefaultOrderId() {
         return 998;
     }
 
+    @Override
+    public boolean doPreUpdateApplication(ServiceProvider serviceProvider,
+                                          String tenantDomain,
+                                          String userName) throws IdentityApplicationManagementException {
+
+        handleWsTrustAssociationRemoval(serviceProvider);
+        return super.doPreUpdateApplication(serviceProvider, tenantDomain, userName);
```

Comment:
since this is extended from the abstract listener, no need to call the super method.